File: chess-core/src/main/kotlin/net/zomis/chess/core/Piece.java
package net.zomis.chess.core;

public enum Piece {

    KING('K'), QUEEN('Q'), ROOK('R'), BISHOP('B'), KNIGHT('N'), PAWN('P');

    private final char ch;

    Piece(char ch) {
        this.ch = ch;
    }

    @Override
    public String toString() {
        return String.valueOf(ch);
    }
}

File: chess-core/src/main/kotlin/net/zomis/chess/core/Piece.kt
package net.zomis.chess.core

enum class Piece private constructor(private val ch: Char) {

    KING('K'), QUEEN('Q'), ROOK('R'), BISHOP('B'), KNIGHT('N'), PAWN('P');

    override fun toString(): String {
        return ch.toString()
    }
}
